### PR TITLE
GitHub action for automatic translations updates

### LINKED ISSUE #4598: Prepare for clang-format 11
This is a heads up, that the `.clang-format` should be reviewed for clang-format 11. The Github actions currently run version 10, but my local version 11 found lots of format changes. To prevent a complete formatting by Github, we should check this, before the runners are updated.
See
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md
https://github.com/actions/virtual-environments/issues/2130

### LINKED ISSUE #4933: Add a check for developers.json
Every time a comma is missing, or a table entry ends with a comma, the website throws lots of server errors.

So a check for validity for this file would be fine.

The website code fails e.g. here https://github.com/widelands/widelands-website/blob/master/mainpage/views.py#L104

### PR DESCRIPTION
**Pull request by Noordfrees**
Suggestion: Instead of manually pulling and pushing the translations occasionally, how about having it done automatically by a GitHub action every night?
@gunchleoc 

If we want this, we should create a Transifex account for bunnybot, make him an admin, and add his TX API token as a secret to the organization for use in this action. A similar action is already implemented and functional in https://github.com/widelands/wl_addons_server/blob/master/.github/workflows/i18n.yaml (it currently uses my personal API token though which I'm not entirely happy about).

---

Given no response for a long time we should just do that.

Additionally some minor other action tweaks which fix #4598 and fix #4933.

### CODE REVIEW

**Comment by Noordfrees** [comment]
Anyone to review this please? We won't get a much more convenient time than now for the formatting change.

The only real changes are in the three YAML files; the rest is formatting only.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -6,17 +6,23 @@ on:
...
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
       uses: actions/checkout@v2
       with:
         fetch-depth: 1
+    - name: Validate JSON files
+      run: |
+        python3 utils/validate_json.py data/txts/
+        python3 utils/update_authors.py
```
Comment (matthiakl) [inline .github/workflows/build.yaml:20]: The outcome of this script is not pushed back into git. Is this intended?

**Review by matthiakl — COMMENTED** [review]
> If we want this, we should create a Transifex account for bunnybot, make him an admin, and add his TX API token as a secret to the organization for use in this action. 

Is this already configured?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (Noordfrees) [reply .github/workflows/build.yaml:20]: The updated files are pushed frequently by the i18n updater workflow already. Here we only check whether any of these programs return an error code in which case the branch author is expected to fix the json syntax.

**Comment by Noordfrees** [comment]
> > If we want this, we should create a Transifex account for bunnybot, make him an admin, and add his TX API token as a secret to the organization for use in this action.
> 
> Is this already configured?

I'll configure this tomorrow morning or this evening

**Review by matthiakl — APPROVED** [review]
Let's have it then.
I'm not really happy about the format changes, but it looks like our old format isn't supported anymore.

**Comment by Noordfrees** [comment]
…should be set up now. We'll see tomorrow morning if it worked.
Thanks for the review :)

**Comment by klaus-halfmann** [comment]
There is a nice tool called jq that can be used for all kind of jsoe manipulations (including validation).
Would this help us?

**Comment by Noordfrees** [comment]
We already have a [json validator](https://github.com/widelands/widelands/blob/master/utils/validate_json.py)
No need to reinvent the wheel ;)

**Pull request merged**